src/python/pants/backend/python/util_rules/pex_environment.py

```diff
@@ -106,21 +106,29 @@ class PexEnvironment(EngineAwareReturnType):
 
     def create_argv(
         self, pex_path: str, *args: str, python: Optional[PythonExecutable] = None
-    ) -> Iterable[str]:
+    ) -> Tuple[str, ...]:
         python = python or self.bootstrap_python
         argv = [python.path] if python else []
         argv.extend((pex_path, *args))
-        return argv
+        return tuple(argv)
 
-    @property
-    def environment_dict(self) -> Mapping[str, str]:
-        return dict(
+    def environment_dict(self, *, python_configured: bool) -> Mapping[str, str]:
```

Comment:
This might seem mildly pedantic, but in a function that’s called in so many places (especially since I proposed adding two other arguments earlier up), I think the adage of “turn every
bool argument into an enum” might be useful to apply here. Keyword-only arguments are definitely half the battle though.

Reply:
Ok, I see you’ve mentioned that in the docstring. With so many callers though (where the docstring is not visible without a fantastic IDE) I feel like the above still holds.

Reply:
As explained above, I think we don't want to add more kwargs here.